2024/02-data-comm/slides.qmd

```diff
@@ -85,33 +89,274 @@ Use `theme_set(theme_bw())` outside your plot code to set the theme for a whole
 -   The `theme()` function allows you to modify individual elements of a theme
 -   The `theme` options are vast. We'll break it down.
 
-## Anatomy of `theme`
+## Anatomy of `theme` {.smaller}
 
 - `theme( <part of the plot> = <element_*(element_options)>)`
 - Parts of the plot: axis, legend, panel, plot, strip.
 - Elements: element_blank, element_text, element_line, element_rect.  
 - You can modify all lines/rectangles at once using `theme(line = element_line())`, etc. 
+- See the [rendered documentation](https://ggplot2.tidyverse.org/reference/theme.html) to help you figure out how to change specific elements.
+
+## Remove the background panel
+
+```{r, echo = T}
+ggplot(penguins, aes(body_mass_g, 
+bill_depth_mm,
+color = species)) +
+geom_point() +
+theme(panel.background = element_blank()) 
+```
+
+## Re-add grid lines
+
+```{r, echo = T}
+ggplot(penguins, aes(body_mass_g, 
+bill_depth_mm,
+color = species)) +
+geom_point() +
+theme(panel.background = element_blank()) +
+theme(panel.grid.major = element_line(color = "black", linewidth = .5))
+
+```
+
+
+
+## Hierarchichal modifications
 
-## Modifying elements
+```{r, echo = T}
+ggplot(penguins, aes(body_mass_g, 
+bill_depth_mm,
+color = species)) +
+geom_point() +
+theme(panel.background = element_blank()) +
+theme(panel.grid.major.x = element_line(color = "black", linewidth = .5)) # Modify only the x axis
```

Comment:
Something like this might make the hierarchical nature and inheritance more obvious.
```suggestion
theme(
  panel.background = element_blank(),
  panel.grid.major = element_line(color = "black", linewidth = 0.5),
  panel.grid.major.x = element_line(linewidth = 0.1)
) 
```